docs/mpi_functions.rst

```diff
@@ -0,0 +1,144 @@
+   ``GlobusMPIEngine`` is required for the partitioning of a batch job (blocks) dynamically
+   based on the ``resource_specification`` of the |MPIFunction|_.
+
+
+3. |MPIFunction|_ automatically prefixes the supplied command with ``$PARSL_MPI_PREFIX``
+   which resolves to an appropriate mpi launcher prefix (for e.g, ``mpiexec -n 4 -host <NODE1,NODE2>``).
+
+.. warning::
+   |MPIFunction|_ allows for multi-line commands, however the MPI launcher prefix is applied only to
+   the entire command. If multiple commands need to be launched, explicitly use the ``$PARSL_MPI_PREFIX``.
+
+Here is an example configuration for Polaris@ALCF:
+
+.. code-block:: yaml
+
+    # Example configuration from Polaris
+    display_name: PolarisMPI@ALCF
+    engine:
+      type: GlobusMPIEngine
+      mpi_launcher: mpiexec
+
+      address:
+        type: address_by_interface
+        ifname: bond0
+
+      provider:
+        type: PBSProProvider
+
+        account: SuperBERT
+        queue: debug-scaling
+
+        # e.g., "#PBS -l filesystems=home:grand:eagle\n#PBS -k doe"
+        scheduler_options: "#PBS -l filesystems=home:grand:eagle"
+
+        # Node setup: activate necessary conda environment and such
+        worker_init: ~/setup_agpt_env.sh
+
+        walltime: 01:00:00
+        nodes_per_block: 4
+
+
+Here's another example for Expanse@SDSC:
+
+.. code-block:: yaml
+
+    # Example configuration from Expanse @ SDSC
+    display_name: ExpanseMPI@SDSC
+    engine:
+      type: GlobusMPIEngine
+      mpi_launcher: srun
+
+      address:
+        type: address_by_interface
+        ifname: ib0
+
+      provider:
+        type: SlurmProvider
+        partition: compute
+        account: <ACCOUNT>
+
+        launcher:
+          type: SimpleLauncher
+
+        # Command to be run before starting a worker
+        # e.g., "module load anaconda3; source activate gce_env"
+        worker_init: <WORKER_INIT>
+        nodes_per_block: 2
+        init_blocks: 0
+        min_blocks: 0
+        max_blocks: 1
+        walltime: 00:05:00
+
+
+.. code-block:: python
+
+   from globus_compute_sdk import MPIFunction
+
+    ep_id = <SPECIFY_ENDPOINT_ID>
+    func = MPIFunction("hostname")
+    for nodes in range(1,4):
+        executor.resource_specification = {
+             "num_nodes": 2,
+             "ranks_per_node": nodes
+        }
+        fu = executor.submit(func)
+        mpi_result = fu.result()
+        print(mpi_result.stdout)
+
+        # The above line prints:
+        exp-14-08
+        exp-14-20
+
+        exp-14-08
+        exp-14-20
+        exp-14-08
+        exp-14-20
+
+        exp-14-08
+        exp-14-20
+        exp-14-08
+        exp-14-08
+        exp-14-20
+        exp-14-20
+
+The |ShellResult|_ object captures outputs relevant to simplify debugging when execution
+failures. By default, |ShellFunction|_ captures 1000 lines of stdout and stderr, but this
+can be changed via the `ShellFunction(snippet_lines)` kwarg.
+
```

Comment:
Is this `` `ShellFunction(snippet_lines)` `` supposed to be a link?  I don't think this is the correct syntax?